In [12]:
import pandas as pd
from akapriori import apriori
import os 

In [2]:
%pwd

'c:\\Users\\surface\\OneDrive\\Desktop\\Instacart_Market_Analysis\\Instacart-Market-Analysis-Project\\Research'

In [3]:
os.chdir('../')

In [4]:
%pwd

'c:\\Users\\surface\\OneDrive\\Desktop\\Instacart_Market_Analysis\\Instacart-Market-Analysis-Project'

In [6]:
order_products_prior_df=pd.read_csv('Instacart_Data/order_products_prior.csv')
products_df=pd.read_csv('Instacart_Data/products.csv')

In [8]:
merged_df = pd.merge(order_products_prior_df, products_df, on='product_id', how='inner')

# Group products by order_id and aggregate into a list
grouped_df = merged_df.groupby('order_id')['product_name'].apply(list).reset_index()

print(grouped_df.head())

   order_id                                       product_name
0         2  [Organic Egg Whites, Michigan Organic Kale, Ga...
1         3  [Total 2% with Strawberry Lowfat Greek Straine...
2         4  [Plain Pre-Sliced Bagels, Honey/Lemon Cough Dr...
3         5  [Bag of Organic Bananas, Just Crisp, Parmesan,...
4         6  [Cleanse, Dryer Sheets Geranium Scent, Clean D...


In [14]:

min_support = 0.01 
min_confidence = 0.2  
# Generate apriori association rules
rules = apriori(grouped_df['product_name'], support=min_support, confidence=min_confidence)

# Print the association rules
for rule in rules:
    print(rule)

(frozenset({'Organic Strawberries'}), frozenset({'Bag of Organic Bananas'}), 0.019169647084147, 0.23283701635541385, 1.9727017264951765)
(frozenset({'Organic Raspberries'}), frozenset({'Organic Strawberries'}), 0.01053322774080726, 0.24707238594161554, 3.0009732007029744)
(frozenset({'Organic Strawberries'}), frozenset({'Banana'}), 0.017467558604163026, 0.21216322922137046, 1.4433528707793088)
(frozenset({'Strawberries'}), frozenset({'Banana'}), 0.012825386002686264, 0.28843449853446285, 1.9622286247214302)
(frozenset({'Organic Raspberries'}), frozenset({'Bag of Organic Bananas'}), 0.012598627504530504, 0.29551938244671927, 2.503775409471641)
(frozenset({'Large Lemon'}), frozenset({'Banana'}), 0.01271589493087443, 0.26778988189208486, 1.8217826727708029)
(frozenset({'Organic Baby Spinach'}), frozenset({'Banana'}), 0.015986629647071704, 0.21244538506371916, 1.4452723855159375)
(frozenset({'Organic Avocado'}), frozenset({'Banana'}), 0.016608738009638947, 0.301982297881967, 2.054394713787

# Rule 9: Organic Fuji Apple -> Banana

Support: 1.06%
Confidence: 37.9%
Lift: 2.576
Percentage of Orders: 1.06%
# Rule 4: Strawberries -> Banana

Support: 1.28%
Confidence: 28.8%
Lift: 1.962
Percentage of Orders: 1.28%
# Rule 1: Organic Strawberries -> Bag of Organic Bananas

Support: 1.92%
Confidence: 23.3%
Lift: 1.973
Percentage of Orders: 1.92%
# Rule 5: Organic Raspberries -> Bag of Organic Bananas

Support: 1.26%
Confidence: 29.6%
Lift: 2.504
Percentage of Orders: 1.26%
# Rule 10: Organic Hass Avocado -> Bag of Organic Bananas

Support: 1.94%
Confidence: 29.2%
Lift: 2.473
Percentage of Orders: 1.94%
# Rule 2: Organic Raspberries -> Organic Strawberries

Support: 1.05%
Confidence: 24.7%
Lift: 3.001
Percentage of Orders: 1.05%
# Rule 6: Large Lemon -> Banana

Support: 1.27%
Confidence: 26.8%
Lift: 1.822
Percentage of Orders: 1.27%
# Rule 7: Organic Baby Spinach -> Banana

Support: 1.60%
Confidence: 21.2%
Lift: 1.445
Percentage of Orders: 1.60%
# Rule 3: Organic Strawberries -> Banana

Support: 1.75%
Confidence: 21.2%
Lift: 1.443
Percentage of Orders: 1.75%
# Rule 8: Organic Avocado -> Banana

Support: 1.66%
Confidence: 30.2%
Lift: 2.054
Percentage of Orders: 1.66%